test/e2e/ingress/sanity.go

```diff
@@ -0,0 +1,62 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ingress
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/api7/ingress-controller/test/e2e/scaffold"
+)
+
+type ip struct {
+	IP string `json:"ip"`
+}
+
+var _ = ginkgo.Describe("single-route", func() {
+	s := scaffold.NewDefaultScaffold()
+	ginkgo.It("/ip should return your ip", func() {
+		backendSvc, backendSvcPort := s.DefaultHTTPBackend()
+		s.CreateApisixRoute("httpbin-route", []scaffold.ApisixRouteRule{
+			{
+				Host: "httpbin.com",
+				HTTP: scaffold.ApisixRouteRuleHTTP{
+					Paths: []scaffold.ApisixRouteRuleHTTPPath{
+						{
+							Path: "/ip",
+							Backend: scaffold.ApisixRouteRuleHTTPBackend{
+								ServiceName: backendSvc,
+								ServicePort: backendSvcPort[0],
+							},
+						},
+					},
+				},
+			},
+		})
```

Comment:
I think we need to use yaml to manage the creation of `ApisixRoute`, In this way, we can simulate the user's usage scenario test conversion process.

This is an enhancement and will not block PR .